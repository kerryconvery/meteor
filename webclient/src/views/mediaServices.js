import axios from 'axios';

export const launchMedia = (profile, uri) => axios.post(`/api/media?profile=${profile}&uri=${uri}`);
export const getMedia = (profile, uri) => axios.get(`/api/profiles/${profile}/media`, { params: { uri } }).then(res => res.data);
export const getProfiles = () => axios.get('/api/profiles').then(res => res.data);
export const pause = () => axios.post('api/media/pause');
export const stop = () => axios.delete('api/media');
export const resume = () => axios.post('api/media/resume');
