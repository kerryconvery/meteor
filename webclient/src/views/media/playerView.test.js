
import React from 'react';
import { shallow } from 'enzyme';
import * as Services from '../mediaServices';
import PlayerView from './playerView';

jest.mock('../mediaServices');

describe('PlayerView', () => {
  const mountComponent = () => shallow(<PlayerView />);

  it('should have an initial state', () => {
    const wrapper = mountComponent();

    expect(wrapper.state()).toEqual({
      offline: true,
      nowPlaying: '',
      playerState: 0,
      position: 0,
      duration: 0,
    });
  });

  it('should start listening for media player events', () => {
    const wrapper = mountComponent();
    const connectSpy = jest.spyOn(wrapper.instance(), 'connectToPlayer');

    wrapper.instance().componentDidMount();
    expect(connectSpy).toHaveBeenCalled();
  });

  it('should render offline button when not online', () => {
    const wrapper = mountComponent();

    expect(wrapper.find('OfflineButton').exists()).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render player when not online', () => {
    const wrapper = mountComponent();

    expect(wrapper.find('MediaPlayer').exists()).toEqual(false);
  });

  it('should not render offline button when online', () => {
    const wrapper = mountComponent();

    wrapper.setState({ offline: false });
    wrapper.update();

    expect(wrapper.find('OfflineButton').exists()).toEqual(false);
  });

  it('should render player when online', () => {
    const wrapper = mountComponent();

    wrapper.setState({ offline: false, playerState: 2 });
    wrapper.update();

    const player = wrapper.find('MediaController');

    expect(player.exists()).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render offline button or player when online but not playing', () => {
    const wrapper = mountComponent();

    wrapper.setState({ offline: false, playerState: 0 });
    wrapper.update();

    const player = wrapper.find('MediaController');

    expect(player.exists()).toEqual(false);
    expect(wrapper.find('OfflineButton').exists()).toEqual(false);
  });

  describe('connectToPlayer', () => {
    it('should call media services listen', () => {
      const wrapper = mountComponent();

      wrapper.instance().connectToPlayer();
      expect(Services.listen).toHaveBeenCalledWith(wrapper.instance().onPlayerUpdated, wrapper.instance().onPlayerDisconnected);
    });
  });

  describe('onPlayerUpdated', () => {
    it('should update the state with player state', () => {
      const payload = {
        nowPlaying: 'movie.avi',
        state: 2,
        position: 100,
        duration: 1000,
      };

      const wrapper = mountComponent();

      wrapper.instance().onPlayerUpdated(payload);
      wrapper.update();

      expect(wrapper.state()).toEqual({
        offline: false,
        nowPlaying: payload.nowPlaying,
        playerState: payload.state,
        position: payload.position,
        duration: payload.duration,
      });
    });
  });

  describe('onPlayerDisconnected', () => {
    it('should set the offline state to true', () => {
      const wrapper = mountComponent();

      wrapper.setState({ offline: false });
      wrapper.instance().onPlayerDisconnected();

      expect(wrapper.state().offline).toEqual(true);
    });
  });
});
