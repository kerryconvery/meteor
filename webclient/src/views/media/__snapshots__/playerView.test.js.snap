// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerView should render offline button when not online 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onClick": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PlayerView should render player when online 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "duration": 0,
      "nowPlaying": "",
      "onPark": [Function],
      "onPause": [MockFunction],
      "onRestart": [Function],
      "onResume": [MockFunction],
      "onStop": [MockFunction],
      "playerState": 2,
      "position": 0,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "duration": 0,
        "nowPlaying": "",
        "onPark": [Function],
        "onPause": [MockFunction],
        "onRestart": [Function],
        "onResume": [MockFunction],
        "onStop": [MockFunction],
        "playerState": 2,
        "position": 0,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
